cluster/etcd_service_discovery_test.go

```diff
@@ -367,3 +404,47 @@ func TestEtcdWatchChangesDeleteServers(t *testing.T) {
 		})
 	}
 }
+
+func TestEtcdWatchChangesWithBlacklist(t *testing.T) {
+	t.Parallel()
+	for _, table := range etcdSDBlacklistTables {
+		t.Run(table.name, func(t *testing.T) {
+			conf := viper.New()
+			conf.Set("pitaya.cluster.sd.etcd.syncservers.interval", "10ms")
+			config := getConfig(conf)
+			c, cli := helpers.GetTestEtcd(t)
+			defer c.Terminate(t)
+			e := getEtcdSD(t, config, table.server, table.serverTypeBlacklist, cli)
+			e.running = true
+			_ = e.bootstrapServer(table.server)
+			e.watchEtcdChanges()
+
+			serversBefore, err := e.GetServersByType(table.server.Type)
+			assert.NoError(t, err)
+			assert.Equal(t, 1, len(serversBefore))
+
+			// Add all servers to ETCD
+			for _, serverToAdd := range table.serversToAdd {
+				err = e.addServerIntoEtcd(serverToAdd)
+				_, err := cli.Put(
+					context.TODO(),
+					getKey(serverToAdd.ID, serverToAdd.Type),
+					serverToAdd.AsJSONString(),
+				)
+				assert.NoError(t, err)
+			}
+
+			// We sleep to guarantee that the servers were added to etcd
+			time.Sleep(time.Millisecond * 50)
```

Comment:
This may make the test flaky, don't you think?

Reply:
The server is in memory, so this should be a very large margin. Do you think something else is required?

Reply:
Oh, the cli.Put method is synchronous, so the sleep is probably unnecessary.

Reply:
there are better ways of waiting in a test, something like that:

https://github.com/topfreegames/pitaya/blob/master/cluster/nats_rpc_client_test.go#L158

let me know what you think

Reply:
other example https://github.com/topfreegames/pitaya/blob/931f2880440b52b045c553bdb0b20502fdba09bc/session/session_test.go#L612

Reply:
there is no simple way of doing that because for some reason `sync.Map` does not have a Len method.  But I'll see a way of doing it.